### PR TITLE
refactor(conf): use DSN_DEFINE_bool to load bool type of configs

### PR DESCRIPTION
**Pull request by acelyc111**
https://github.com/apache/incubator-pegasus/issues/1323

This patch refactors the code to use `DSN_DEFINE_bool` instead of `dsn_config_get_value_bool` to load bool type of configurations, and doesn't introduce any functional changes.
- all default value and most of description are kept as before
- move the defination of flags closer to the places where they're used

### CODE REVIEW

File: src/utils/simple_logger.cpp [hunk 1]
```diff
@@ -40,7 +40,6 @@ namespace dsn {
 namespace tools {
 
 DSN_DEFINE_bool(tools.simple_logger, fast_flush, false, "whether to flush immediately");
-
 DSN_DEFINE_bool(tools.simple_logger,
                 short_header,
```
Comment (empiredan) [inline src/utils/simple_logger.cpp:44]: Add another `short_header` for `tools.screen_logger` ?

File: src/replica/replica_stub.cpp [hunk 2]
```diff
@@ -719,7 +752,7 @@ void replica_stub::initialize(const replication_options &opts, bool clear /* = f
     }
 
     // gc
-    if (false == _options.gc_disabled) {
+    if (false == FLAGS_gc_disabled) {
```
Comment (empiredan) [inline src/replica/replica_stub.cpp:755]:
```suggestion
    if (!FLAGS_gc_disabled) {
```

File: src/utils/simple_logger.cpp (hunk 1 above)
Comment (acelyc111) [reply src/utils/simple_logger.cpp:44]: It seems `screen_logger(const char *log_dir)` is only used in one test, and the parameter `log_dir` is useless, so I removed the c'stor and updated the only caller.

ps: Two sections with one same key is not implemented yet.

Comment (empiredan) [reply src/utils/simple_logger.cpp:44]: OK, that means `[tools.screen_logger].short_header` is not used anywhere.

**Review by empiredan — APPROVED** [review]

**Review by levy5307 — APPROVED** [review]

**Pull request merged**